### PR TITLE
Use protocol from `this._ws.protocol` only if it is defined

### LINKED ISSUE #8: Error: Cannot read property 'split' of undefined
On React Native, this error occurs after successfully connecting to the server:
```
Error: Cannot read property 'split' of undefined
 stack: 
  Wampy._wsOnOpen                              index.ios.bundle:58797
  WebSocket.Wampy._initWsCallbacks._ws.onopen  index.ios.bundle:58785
  WebSocket.<anonymous>                        index.ios.bundle:13888
  Object.ErrorUtils.applyWithGuard             index.ios.bundle:1202
  EventEmitter.emit                            index.ios.bundle:9702
  MessageQueue.__callFunction                  index.ios.bundle:7157
  <unknown>                                    index.ios.bundle:7093
  guard                                        index.ios.bundle:7046
  <unknown>                                    index.ios.bundle:7093
 URL: undefined
 line: undefined
 message: Cannot read property 'split' of undefined
```

This is because the React Native [WebSocket iOS implementation does not currently support passing the 'protocols' option in the constructor](https://github.com/facebook/react-native/blob/babdc2161445f839e76c35365dfa35b199d60dbe/Libraries/WebSocket/WebSocket.ios.js#L25), so [`this._ws.protocol`](https://github.com/KSDaemon/wampy.js/blob/d09c2581f7efff190f61299a890367c528cb7d14/src/wampy.js#L655) ends up being undefined.

### PR DESCRIPTION
**Pull request by elliottsj**
Fixes #8

### CODE REVIEW

File: src/wampy.js [hunk 1]
```diff
@@ -652,8 +652,10 @@
 
         this._log('[wampy] websocket connected');
 
-        p = this._ws.protocol.split('.');
-        this._options.transportEncoding = p[2];
+        if (this._ws.protocol) {
+            p = this._ws.protocol.split('.')[2];
+        }
+        this._options.transportEncoding = p || this._options.transportEncoding;
```
Comment (KSDaemon) [inline src/wampy.js]: This all can be optimized to:
```JavaScript
        if (this._ws.protocol) {
            this._options.transportEncoding = this._ws.protocol.split('.')[2];
        }
```
and no need in local var p ;)

**Comment by KSDaemon** [comment]
Hi, @elliottsj ! Thanks for you issue/PR! Do you use wampy in some application, build with React Native?

**Comment by elliottsj** [comment]
Hey @KSDaemon. Yep, I'm trying out wampy for a React Native app I'm building, not open source (yet).

File: src/wampy.js (hunk 1 above)
Comment (elliottsj) [inline src/wampy.js]: Good catch, I'll fix shortly

Comment (elliottsj) [inline src/wampy.js]: Fixed!

**Pull request merged**